usaspending_api/search/v2/views/search.py

```diff
@@ -150,307 +155,249 @@ def post(self, request):
 
         # filter the transactions by category
         if category == "awarding_agency":
-            # TODO: Add "offices" below for office_agency changes
             potential_scopes = ["agency", "subagency"]
             if scope not in potential_scopes:
                 raise InvalidParameterException("scope does not have a valid value")
-            # filter the transactions by scope name
-            name_dict = {}  # {ttname: {aggregated_amount: 1000, abbreviation: "tt"}
-            # define what values are needed in the sql query
 
             if scope == "agency":
-                agency_set = queryset.values('awarding_agency__toptier_agency__name', 'awarding_agency__toptier_agency__abbreviation') \
-                        .annotate(federal_action_obligation=Sum('federal_action_obligation'))\
-                        .order_by('-federal_action_obligation')
-
-                agency_set = agency_set[lower_limit:upper_limit + 1]
-                has_next = len(agency_set) > limit
-                has_previous = page > 1
-
-                for trans in agency_set[:limit]:
-                    ttname = trans["awarding_agency__toptier_agency__name"]
-                    ttabv = trans["awarding_agency__toptier_agency__abbreviation"]
-                    ttob = trans["federal_action_obligation"] if trans["federal_action_obligation"] else 0
-                    if ttname in name_dict:
-                        name_dict[ttname]["aggregated_amount"] += ttob
-                    else:
-                        name_dict[ttname] = {"aggregated_amount": ttob, "abbreviation": ttabv}
+                agency_set = queryset \
+                    .filter(awarding_agency__toptier_agency__name__isnull=False) \
+                    .values(
+                        agency_name=F('awarding_agency__toptier_agency__name'),
+                        abbreviation=F('awarding_agency__toptier_agency__abbreviation')) \
+                    .annotate(aggregated_amount=Sum('federal_action_obligation')) \
+                    .order_by('-aggregated_amount')
+
+                # Begin DB hits here
+                total = agency_set.count()
```

Comment:
Regarding the first comment, here's an example and really it applies to most the cases (see lines 185, 237, 252, 298 specifically). Is there a reason we're not using the simple pagination and using the `count`s?

Reply:
I was focusing on refactoring spending_by_category. I didn't want to change the Gist. count is listed in the gist file